src/relations/natsclient_provider.py

```diff
@@ -77,13 +78,12 @@ def expose_nats(self, auth_token=None):
             # TODO: make this the only way to share url once all charms use the
             # charm-lib and juju > 3.0
             if JujuVersion.from_environ().has_secrets:
+                secret_label = f"{NATS_URL_SECRET_LABEL_PREFIX}{self._charm.unit.name}"
                 try:
-                    secret = self.model.get_secret(label=self._secret_label)
```

Comment:
@morphis  @jat-canonical 
I'm not sure if there are any existing NATS clients in the wild using the secret to fetch the NATS URL. If so, this would be a breaking change, could you help confirm this?  
Thanks.

Reply:
There might be, so we have to stay backward compatible